### PR TITLE
Add android.jar to the searchable jars list for error checking stuffs

### PR DESCRIPTION
**Pull request by omerjerk**

### CODE REVIEW

File: src/processing/mode/android/AndroidMode.java [hunk 1]
```diff
@@ -157,6 +157,20 @@ public void checkSDK(Editor parent) {
...
+  
+  
+  @Override
+  public String getSearchPath() {
+    StringBuilder tehPath = new StringBuilder(super.getSearchPath());
+    tehPath.append(File.pathSeparator);
+
+    String androidJarPath = sdk.getSdkFolder().getAbsolutePath() + "/platforms/"
+    	  + "android-" + AndroidBuild.sdkVersion + "/android.jar";
+    System.out.println("androidJarPath = " + androidJarPath);
```
Comment (benfry) [inline src/processing/mode/android/AndroidMode.java]: Accidental debug string?

Comment (omerjerk) [inline src/processing/mode/android/AndroidMode.java]: Oh sorry. Pushing the fixed commit over it.

File: src/processing/mode/android/AndroidMode.java [hunk 2]
```diff
@@ -157,6 +157,20 @@ public void checkSDK(Editor parent) {
   public AndroidSDK getSDK() {
     return sdk;
   }
+  
+  
+  @Override
+  public String getSearchPath() {
+    StringBuilder tehPath = new StringBuilder(super.getSearchPath());
```
Comment (benfry) [inline src/processing/mode/android/AndroidMode.java]: No sense in using StringBuilder if you're just concatenating all the strings passed to it. Just do `return super.getSearchPath() + [everything else]`

Comment (omerjerk) [inline src/processing/mode/android/AndroidMode.java]: Okay,

**Comment by Manindra29** [comment]
@omerjerk, a couple of things I noticed:

If the searchPath contains 2 files: FileA, FileB, I believe they should be separated by a  : , as we do it in the system PATH environment variable.

`String searchPath =  FileA.getAbsolutePath() + File.pathSeparatorChar + FileB.getAbsolutePath();`

I think it should be `return super.getSearchPath() + File.pathSeparatorChar + androidJarPath;`

Also it's a bad idea to add / char in the path string. Please use the system independent [File.pathSeparator](http://docs.oracle.com/javase/7/docs/api/java/io/File.html#pathSeparator)

**Comment by codeanticode** [comment]
@Manindra29 I applied the changes you suggested, but the problem is that sdk is null when getSearchPath() is called. @omerjerk any ideas?

**Comment by Manindra29** [comment]
Can't we initilalise it or get its value before getSearchPath() is called? @omerjerk

**Comment by omerjerk** [comment]
@codeanticode do you mean this sdk variable is null ? https://github.com/processing/processing-android/blob/master/src/processing/mode/android/AndroidMode.java#L164

I have tested a lot of time but haven't got NPE here at all.
I'm getting NPE [here](https://github.com/processing/processing/blob/master/java/src/processing/mode/java/pdex/ASTGenerator.java#L3307). The classPath variable is somehow null.

**Comment by omerjerk** [comment]
Also, what I do is I've created a test sketch having that `android.view.MotionEvent` import. I open the IDE via `ant run`, then go to File->Open Recent->test sketch.
I never get sdk variable null this way.

**Comment by codeanticode** [comment]
Hi Umair,

I added the following code in the getSearchPath() method:

```java
  public String getSearchPath() {
    System.out.println("in getSearchPath");
    System.out.println("  SDK: " + sdk);
    String androidJarPath = sdk.getSdkFolder().getAbsolutePath() + 
      File.separator + "platforms" + File.separator + "android-" + 
      AndroidBuild.sdkVersion + File.separator + "android.jar";    
    return super.getSearchPath() + File.pathSeparatorChar + androidJarPath;
  }
```

and I see the following in the terminal when I switch to the android mode from the PDE:

```
in getSearchPath
  SDK: null
Exception in thread "Thread-12" java.lang.NullPointerException
	at processing.mode.android.AndroidMode.getSearchPath(AndroidMode.java:166)
	at processing.mode.java.pdex.ASTGenerator.loadJars(ASTGenerator.java:310)
	at processing.mode.java.pdex.ErrorCheckerService$1.run(ErrorCheckerService.java:876)
	at java.lang.Thread.run(Thread.java:745)
```

I'm using the latest revision from Processing.

I also saved a minimal sketch with only the following code:

```java
import android.view.MotionEvent;
void mouseDragged() {
 MotionEvent me = (MotionEvent) mouseEvent.getNative();
}
```

Every time I open it I get the following error in the PDE console:

```text
java.lang.NullPointerException
	at processing.mode.java.pdex.ASTGenerator.getSuggestImports(ASTGenerator.java:3307)
	at processing.mode.java.pdex.ErrorCheckerService.updateErrorTable(ErrorCheckerService.java:940)
	at processing.mode.java.pdex.ErrorCheckerService.checkCode(ErrorCheckerService.java:492)
	at processing.mode.java.pdex.ErrorCheckerService.run(ErrorCheckerService.java:290)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by Manindra29** [comment]
I get a null SDK when launching Processing from eclipse. Running the .app directly doesn't show SDK to be null. Any ideas what we might be missing?

**Comment by omerjerk** [comment]
Okay. I'm also getting null sdk when I run from eclipse. Thanks @Manindra29 :)
Let me see what is going wrong.

**Pull request merged**